File: Leetcode/922/922.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        int n = (int)nums.size();
        int j = 1;
        for (int i = 0; i < n; i += 2) {
            if (nums[i] & 1) {
                for (; j < n; j += 2) {
                    if (!(nums[j] & 1)) {
                        swap(nums[i],nums[j]);
                        break;
                    }
                }
            }
        }
        return nums;
    }
};

File: Leetcode/922/922.go
func sortArrayByParityII(nums []int) []int {
	j := 1
	for i := 0; i < len(nums); i += 2 {
		if (nums[i] & 1) != 0 {
			for ; j < len(nums); j += 2 {
				if nums[j] & 1 == 0 {
					break
				}
			}
			nums[i], nums[j] = nums[j], nums[i]
		}
	}
	return nums
}